packages/hub/services/file-cache.ts

```diff
@@ -160,7 +165,19 @@ export default class FileCache {
       }
       let context = {
         exports: {},
-        require: (specifier: string) => this.loadModule(specifier),
+        require: (specifier: string) => {
+          // TODO: We need to figure out a pattern for importing npm code
```

Comment:
i  can only see this growing. perhaps we want to create an object whose keys are the specifier values are the exported functions, and have a separate module whose whole job is to reexport allowed npm deps into a specifier based POJO. and then in this require we would check to see if the specifier matches a key in that POJO and return it's value, or otherwise call `this.loadModule(specifier)` if the specifier key doesn't exist in the allowed npm deps POJO

Reply:
Yeah, some basic structure like that seems like a good idea. We currently have another place where we need to do that module tracking manually, the compiled package.json.